concourse/pipelines/guest-package-build.jsonnet

```diff
@@ -164,22 +257,79 @@ local promotepackagejob = {
   ] else [],
   on_success: {
     task: 'success',
-    file: 'guest-test-infra/concourse/tasks/publish-job-result.yaml',
-    vars: {
-      pipeline: 'guest-package-build',
-      job: tl.name,
-      result_state: 'success',
-      start_timestamp: '((.:start-timestamp-ms))',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'registry-image',
+        source: { repository: 'gcr.io/compute-image-tools/daisy' },
+      },
+      run: {
+        path: '/publish-job-result',
+        args: [
+          '--project-id=gcp-guest',
```

Comment:
Adding to your own comments elsewhere, this block could also become a shell script (it's repeated twice at least, and would reduce maintenance later).

Reply:
I'd say adding a TODO for now would be ok

Reply:
+1; every one of these only differs by tl.package. You could make this a construct at the top that gets re-used? Or the construct could be re-used across pipelines (since I expect other pipelines to be very similar in structure needs to the job metric gathering.

Reply:
this one is a bit different than the ones where i suggested that. those ones have newlines being passed as one big string to a shell interpreter. this one has discrete arguments being passed to a binary. that said, i could extract this job to a template for reuse

Reply:
extracted to template!

Reply:
great! thanks.